src/scripts/main.js

```diff
@@ -1,3 +1,239 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('table');
+const tbody = document.querySelector('table tbody');
+const wrapperForm = document.createElement('div');
+
+wrapperForm.style.cssText = `display: flex; flex-direction: column; row-gap: 20px; position: relative`;
+
+// --------------------SORTING OF TABLE
+table.addEventListener('click', (e) => {
+  const target = e.target.closest('thead th');
+
+  if (!target) {
+    return;
+  }
+
+  setAttr(target);
+
+  sort(target);
+});
+
+// CHANGE STYLES FOR CURRENT ROW
+table.addEventListener('click', (e) => {
+  const target = e.target.closest('tbody tr');
+
+  if (!target) {
+    return;
+  }
+
+  [...tbody.rows].forEach((el) => {
+    el.classList.remove('active');
+  });
+
+  target.classList.add('active');
+});
+
+// ---------------------ATTRIBUTE SETTING TO CONTROL SORTING
+function setAttr(elem) {
+  let currentAttr = null;
+
+  if (!elem.hasAttribute('data-sort')) {
+    currentAttr = 'asc';
+  } else {
+    currentAttr = elem.dataset.sort === 'asc' ? 'desc' : 'asc';
+  }
+
+  const thead = [...table.rows[0].cells];
+
+  thead.forEach((el) => {
+    el.removeAttribute('data-sort');
+  });
+
+  elem.setAttribute('data-sort', currentAttr);
+}
+
+//  -------------------------SORTING ELEMENTS
+function sort(elem) {
+  // WE FIND THE COLUMN INDEX NUMBER
+  const thead = [...table.rows[0].cells];
+  const currentPos = thead.indexOf(elem);
```

Comment:
The `indexOf` method will return -1 if the element is not found. It's a good practice to check if `currentPos` is not -1 before proceeding with sorting to avoid potential errors.